Mathlib/Algebra/Polynomial/Roots.lean

```diff
@@ -751,6 +751,16 @@ theorem roots_map_of_injective_of_card_eq_natDegree [IsDomain A] [IsDomain B] {p
   apply Multiset.eq_of_le_of_card_le (map_roots_le_of_injective p hf)
   simpa only [Multiset.card_map, hroots] using (card_roots' _).trans (natDegree_map_le f p)
 
+theorem roots_map_of_map_ne_zero_of_card_eq_natDegree [IsDomain A] [IsDomain B] {p : A[X]}
+    {f : A →+* B} (h : p.map f ≠ 0) (hroots : p.roots.card = p.natDegree) :
```

Comment:
I suggest keeping `f` explicit. Because probably `h` is not yet floating around in the context during applications, and should become a new goal.